priority/priority.erl

```diff
@@ -1,47 +1,45 @@
 -module(priority).
+
 -export([start/0,
          send_vip/2,
          send_normal/2]).
+
  -include_lib("eunit/include/eunit.hrl").
 
+start() ->
+    spawn(fun() -> priority_loop() end).
+
+send_vip(Pid, Msg) ->
+    Pid ! {vip, Msg}.
+
+send_normal(Pid, Msg) ->
+    Pid ! {normal, Msg}.
+
 priority_process_msg(Msg) ->
-    io:format("Processing importante msg: ~p~n",[Msg]),
+    io:format("Processing importante msg: ~p~n", [Msg]),
     timer:sleep(500). % only for being able to produce faster than cusume (testing...)
 
-priority_loop_vip(State) ->
-    receive 
+priority_loop() ->
+    receive
         {vip,Msg} ->
             priority_process_msg(Msg),
-            priority_loop_vip(State)
+            priority_loop()
     after 0 ->
-            priority_loop(State)
+            receive 
+                {normal, {Pid, exit}} ->
+                    Pid ! exit;
+                {normal, Msg} ->
+                    io:format("normal msg : ~p~n", [Msg]),
```

Comment:
you are not processing it here. You could actually Match against any kind of priority at this point